federation/pkg/federation-controller/dns/BUILD

```diff
@@ -0,0 +1,33 @@
+package(default_visibility = ["//visibility:public"])
+
+licenses(["notice"])
+
+load(
+    "@io_bazel_rules_go//go:def.bzl",
+    "go_library",
+    "go_test",
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = ["dns_test.go"],
+    library = ":go_default_library",
+    tags = ["automanaged"],
+    deps = [
+        "//federation/apis/federation/v1beta1:go_default_library",
+        "//federation/pkg/dnsprovider/providers/google/clouddns:go_default_library",
+        "//pkg/api/v1:go_default_library",
+        "//pkg/util/sets:go_default_library",
+    ],
+)
+
+go_library(
+    name = "go_default_library",
+    srcs = ["dns.go"],
```

Comment:
That's a good point.  Even then, conceptually I'd prefer something like a separate "service-dns" controller and "ingress-dns" controller, which share common library code where applicable.  I'm not totally sure to what extent that code would be better than a combined controller for both of them though, but architecturally it makes more sense (looser coupling).